src/server/core.rs

```diff
@@ -29,7 +29,7 @@ pub struct Core {
 
     maximal_task_id: TaskId,
     worker_id_counter: WorkerId,
-    scatter_counter: usize,
+    resource_names: Vec<String>, // "Map" from GenericResourceId -> to its name
```

Comment:
Why isn't this a map?

Reply:
It is actually a kind of map, as long as resource_names are indexed from 0 in a dense way.